tests/resources/deeprec/movielens/ML-100K2Libffm.py

```diff
@@ -0,0 +1,209 @@
+                final_str = rating + " " + user_feature_dict[userid] + " " + item_feature_dict[itemid] + "\n"
+                train_final.write(final_str)
+        ## 2-class classification  rating>3 -> class=1
+        elif flag == 'classification':
+
+            for line in train:
+                line = line.strip().split("\t")
+                rating = line[2]
+                userid = line[0]
+                itemid = line[1]
+
+                if int(rating) > 3:
+                    rating = "1.00"
+                else:
+                    rating = "0.00"
+
+                final_str = rating + " " + user_feature_dict[userid] + " " + item_feature_dict[itemid] + "\n"
+                train_final.write(final_str)
+        elif flag =='classification_topN':
+            
+            USER_COL='userID'
+            ITEM_COL='itemID'
+            RATING_COL='rating'
+            TS_COL = 'timestamp'
+            
+            ## top_N recommendation
+            f_set_list = []
+            for i in range(field_num):#{0,1}
+                f_set_list.append(set())
+
+            for line in train:
+                line = line.strip().split("\t")
+                for i in range(0, field_num ): # i"0,1"
+                    f_set_list[i].add(line[i]) ##f_set_list[0] is user and f_set_list[1] is movie id
+
+            user_df = pd.DataFrame(list(f_set_list[0]),columns=[USER_COL])
+            item_df = pd.DataFrame(list(f_set_list[1]),columns=[ITEM_COL])
+            user_items = user_item_pairs(user_df,item_df,user_col=USER_COL,item_col=ITEM_COL)
+
+            user_items[USER_COL] = user_items[USER_COL].astype(int)
+            user_items[ITEM_COL] = user_items[ITEM_COL].astype(int)
+
+            data = pd.read_csv(filepath, sep='\t', names=[USER_COL, ITEM_COL, RATING_COL, TS_COL])
+
+           ## merge all user_item data with movielens data
+            user_item_rating= pd.merge(user_items,data, how='left', on=[USER_COL,ITEM_COL]).fillna(0).drop(columns=[TS_COL])
+            
+            user_item_rating.drop(columns=[RATING_COL]).to_csv('user_item.csv', sep='\t') ##order is a little weird need to check
+            
+            print(len(user_item_rating))
+            #user_item_rating.columns = ['']*len(user_item_rating.columns) #remove header
+
+            for line in range(len(user_item_rating)):
+
+                rating = user_item_rating.iloc[line,2]
+                #print(user_item_rating.loc[1,"userID"])
+                #(user_item_rating.iloc[1,1])
+                userid = user_item_rating.iloc[line,0].astype(str)
+                itemid = user_item_rating.iloc[line,1].astype(str)
+
+                if int(rating) > 3:
+                    rating = "1.00"
+                else:
+                    rating = "0.00"
+
+                final_str = rating + " " + user_feature_dict[userid] + " " + item_feature_dict[itemid] + "\n"
+                train_final.write(final_str)
+
+
+
+if __name__ == "__main__":
```

Comment:
in the other utilities we are not using mains, if we want to test anything, we use the test